drivers/clock_control/clock_stm32_ll_u5.c

```diff
@@ -386,149 +313,167 @@ void config_src_sysclk_hse(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
+	if (IS_ENABLED(STM32_LSE_ENABLED)) {
+		/* Enable the power interface clock */
+		LL_AHB3_GRP1_EnableClock(LL_AHB3_GRP1_PERIPH_PWR);
 
-	/* If freq not increased, set flash latency after all clock setting */
-	if (new_hclk_freq <= old_hclk_freq) {
-		LL_SetFlashLatency(new_hclk_freq);
-	}
+		if (!LL_PWR_IsEnabledBkUpAccess()) {
+			/* Enable write access to Backup domain */
+			LL_PWR_EnableBkUpAccess();
+			while (!LL_PWR_IsEnabledBkUpAccess()) {
+				/* Wait for Backup domain access */
+			}
+		}
 
-	/* Disable other clocks */
-	LL_RCC_HSI_Disable();
-	LL_RCC_MSIS_Disable();
-	LL_RCC_PLL1_Disable();
+		/* Enable LSE Oscillator */
+		LL_RCC_LSE_Enable();
+		/* Wait for LSE ready */
+		while (!LL_RCC_LSE_IsReady()) {
+		}
 
-#endif	/* STM32_SYSCLK_SRC_HSE */
-}
+		/* Enable LSESYS additionnally */
+		LL_RCC_LSE_EnablePropagation();
+		/* Wait till LSESYS is ready */
+		while (!LL_RCC_LSESYS_IsReady()) {
+		}
 
-/*
- * Configure MSI as source of SYSCLK
- */
-void config_src_sysclk_msis(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
-{
-#ifdef STM32_SYSCLK_SRC_MSIS
+		LL_PWR_DisableBkUpAccess();
+	}
 
-	uint32_t old_hclk_freq;
-	uint32_t new_hclk_freq;
+	if (IS_ENABLED(STM32_MSIS_ENABLED)) {
+		/* Set MSIS Range */
+		LL_RCC_MSI_EnableRangeSelection();
 
-	old_hclk_freq = HAL_RCC_GetHCLKFreq();
+		LL_RCC_MSIS_SetRange(STM32_MSIS_RANGE << RCC_ICSCR1_MSISRANGE_Pos);
 
-	/* Calculate new SystemCoreClock variable with MSI freq */
-	/* MSI freq is defined from RUN range selection */
-	new_hclk_freq =	__LL_RCC_CALC_HCLK_FREQ(
-				__LL_RCC_CALC_MSIS_FREQ(LL_RCC_MSIRANGESEL_RUN,
-				STM32_MSIS_RANGE << RCC_ICSCR1_MSISRANGE_Pos),
-				s_ClkInitStruct.AHBCLKDivider);
+		if (IS_ENABLED(STM32_MSIS_PLL_MODE)) {
+			__ASSERT(STM32_LSE_ENABLED,
+				"MSIS Hardware auto calibration needs LSE clock activation");
+			/* Enable MSI hardware auto calibration */
+			LL_RCC_MSI_EnablePLLMode();
+		}
 
-	__ASSERT(new_hclk_freq == CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC,
-			 "Config mismatch HCLK frequency %u %u",
-			 CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC, new_hclk_freq);
+		/* Enable MSIS */
+		LL_RCC_MSIS_Enable();
 
-	/* If freq increases, set flash latency before any clock setting */
-	if (new_hclk_freq > old_hclk_freq) {
-		LL_SetFlashLatency(new_hclk_freq);
+		/* Wait till MSIS is ready */
+		while (LL_RCC_MSIS_IsReady() != 1) {
+		}
 	}
 
-	set_regu_voltage(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
+	if (IS_ENABLED(STM32_MSIK_ENABLED)) {
+		/* Set MSIK Range */
+		LL_RCC_MSI_EnableRangeSelection();
 
-	/* Set MSIS as SYSCLCK source */
-	set_up_clk_msis();
-	LL_RCC_SetSysClkSource(LL_RCC_SYS_CLKSOURCE_MSIS);
-	LL_RCC_SetAHBPrescaler(s_ClkInitStruct.AHBCLKDivider);
-	while (LL_RCC_GetSysClkSource() != LL_RCC_SYS_CLKSOURCE_STATUS_MSIS) {
-	}
+		LL_RCC_MSIK_SetRange(STM32_MSIK_RANGE << RCC_ICSCR1_MSIKRANGE_Pos);
 
-	/* Update SystemCoreClock variable */
-	LL_SetSystemCoreClock(new_hclk_freq);
+		if (IS_ENABLED(STM32_MSIK_PLL_MODE)) {
```

Comment:
according to the  [ref man](https://www.st.com/resource/en/reference_manual/rm0456-stm32u575585-armbased-32bit-mcus-stmicroelectronics.pdf)  the MSIS is the source for the PLL. However if MSIK is selected, there is a bit MSIPLLSEL in the RCC->CR to reset  with `LL_RCC_SetMSIPLLMode()`

Reply:
MSIPLLMode is a MSI clk calibration feature. This is different from MSI being PLL source